exercise3/src/camera_demo/src/actuation_node.cpp

```diff
@@ -0,0 +1,93 @@
+// Listen to the /stop topic and stop the robot when a message is received
+
+#include <chrono>
+#include <memory>
+#include <string>
+#include <vector>
+#include <list>
+#include <thread>
+#include <cactus_rt/tracing.h>
+
+#include "rclcpp/experimental/fifo_sched.hpp"
+#include "rclcpp/rclcpp.hpp"
+#include "std_msgs/msg/empty.hpp"
+
+using namespace std::chrono_literals;
+
+using FileSink = cactus_rt::tracing::FileSink;
+using TraceAggregator = cactus_rt::tracing::TraceAggregator;
+using ThreadTracer = cactus_rt::tracing::ThreadTracer;
+
+
+class ActuationNode : public rclcpp::Node
+{
+public:
+  ActuationNode()
+  : Node("actuation_node")
+  {
+    StartTracing("actuation_node", "actuation_node.json");
+    trace_aggregator = nullptr;
```

Comment:
This lines seems to delete the global `trace_aggregator` which may cause segfaults and/or issues.

Reply:
The trace_aggregator is allocated in the StartTracing function, after this constructor has been called. Nothing has been allocated yet at this point in the code.

Reply:
Correction: the lines are indeed backwards